proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -12,7 +12,7 @@ Swift allows the use of a protocol as a value type when its *requirements* meet
 
 ## Motivation
 
-In Swift, an *existential type* for a given *protocol type* or *protocol composition type* is a formally distinct type that has an equivalent spelling and can hold a value of any conforming type, exposing just the interface(s) of the specified protocol(s) (and superclass). An *existential value* is therefore a value of existential type. The ability to represent an instance of any conforming type enables users and, more importantly, library authors to save on API surface and the considerable amount of boilerplate and difficult-to-maintain code often entailed in the design of ergonomic and exhaustive type-erased wrappers that stand in for a particular existential type. Likewise, existential types can prove incredibly useful in specific use cases that involve dynamicity.
+In Swift, an *existential type* for a given *protocol type* or *protocol composition type* is a formally distinct type that has an equivalent spelling and can hold a value of any conforming type, exposing just the interface(s) of the specified protocol(s) (and superclass). An *existential value* is therefore a value of an existential type. The ability to represent an instance of any conforming type enables users and, more importantly, library authors to save on API surface and the considerable amount of boilerplate and difficult-to-maintain code often entailed in the design of ergonomic and exhaustive type-erased wrappers that stand in for a particular existential type. Likewise, existential types can prove incredibly useful in specific use cases that involve dynamicity.
```

Comment:
This was not a typo –– I cannot speak to whether using an article is correct or not, but have never come across this particular construct in literature. Have a look at the intro in [this](https://en.m.wikipedia.org/wiki/Value_type_and_reference_type) wiki page.
```suggestion
In Swift, an *existential type* for a given *protocol type* or *protocol composition type* is a formally distinct type that has an equivalent spelling and can hold a value of any conforming type, exposing just the interface(s) of the specified protocol(s) (and superclass). An *existential value* is therefore a value of existential type. The ability to represent an instance of any conforming type enables users and, more importantly, library authors to save on API surface and the considerable amount of boilerplate and difficult-to-maintain code often entailed in the design of ergonomic and exhaustive type-erased wrappers that stand in for a particular existential type. Likewise, existential types can prove incredibly useful in specific use cases that involve dynamicity.
```

Reply:
I see. I'm okay with "value of existential type", but I'm wondering if "bound to an existential type" would be more in-line with Swift's terminology. I say that because –– at least I –– mostly say "value of type Int" or "value of existential type P". Here, although correctly used, it seems slightly odd to me as it refers to an abstract existential type. 

If you don't like "bound to" go ahead and commit your suggestion.

Reply:
> it refers to an abstract existential type

That is why this sentence follows the definition for an existential type. It isn't that I don't like the word "bound" -- I haven't seen it used between types and values. In Swift, "bound" is usually used to describe relations between type parameters and other types (memory and types as well, in the stdlib [Pointer](https://developer.apple.com/documentation/swift/unsafepointer/2430863-withmemoryrebound) APIs). Does "some existential type" feel a compromise to you?